### PR TITLE
Update zlib to 1.2.8

### PR DESCRIPTION
**Pull request by davidlt**
Update zlib to 1.2.8 released in Apr 2013. This is a battle tested
version which is used as a base for QAT and Cloudflare zlib forks.

roottest passed with no additional issues.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build please.

**Comment by pcanal** [comment]
@phsft-bot build please.

**Comment by pcanal** [comment]
@phsft-bot build please.

**Comment by amadio** [comment]
@davidlt Is this version still desirable, or should we make the jump directly to a more recent version like 1.2.11?

**Comment by davidlt** [comment]
TL;DR Yes, this is still desirable version.

The 1.2.8 was picked because HW compression/decompression accelerators provide zlib-shim based  on 1.2.8 (Intel and Cavium does that). Also Cloudflare/CMS forks of zlib optimized for SSE4.2, AVX, AVX2 are based on 1.2.8.

**Comment by amadio** [comment]
@davidlt Ok, thanks for the explanation. Are the forks by CMS and Cloudfare only adding vectorization? Could we just upstream those patches if that's the case? Alternatively, should we include those patches in ROOT's version of zlib? zlib is probably quite stable between versions, maybe we can just rebase the patches on the more recent version?

**Comment by davidlt** [comment]
CMS fork is based on Cloudflare fork (which is Intel-only, adds SSE4.2and PCLMUL. CMS adds additional vectorization bits (adler32) and makes it fat (i.e. picks implementations at load time based  on available ISA). You can build it on 10+ year old Intel system and successfully use AVX optimized adler32 if AVX is available at run-time.

This would be introduced as a patch on top (after this is merged). 

There is no effort to rebase Cloudflare fork on 1.2.11 and thus no CMS effort also (as of now).

**Comment by amadio** [comment]
@peremato Could you please let us know what is your opinion? I think that if the update passes the tests, it shouldn't be a problem to merge this in.

@davidlt It would still be interesting to try to upstream the patches to the official library instead of relying on a patched version. Do you think that would be too difficult?

**Comment by amadio** [comment]
@phsft-bot build

**Comment by davidlt** [comment]
TL;DR Difficult due to many reasons.

I wouldn't count on upstreaming this. The upstream very rarely makes new releases. It was 4 years between 1.2.8 and 1.2.9. There are 81 pull request pending and I don't see comments from maintainer.

Plus as-is it's not good enough for upstreaming. There are other pull request adding NEON/ARM optimized pieces thus a different multi-architecture & multi-ISA-extensions design might be needed.

**Comment by amadio** [comment]
The build seems alright, so upon receiving a green light from @peremato we can merge this.

It does look like the maintainer is back, since there were 3 releases this year. Maybe if the patches are reworked they could be upstreamed, which would benefit eveyone. However, for now we should just focus on ROOT and get the base 1.2.8 version merged in.

**Comment by bbockelm** [comment]
Catching up on this one -- any news getting this merged?

**Comment by amadio** [comment]
@bbockelm Hi Brian, I am working on maybe using subtrees to manage the in-source external projects like zlib. However, that involves quite a few changes to our build system, so it will still take a bit until we can have the new version merged.

**Comment by pcanal** [comment]
@amadio I don't think that tying this PR to a completely different one (about subtree) is good idea (especially since the subtree PR is likely to spark some debate).

**Comment by amadio** [comment]
@pcanal We want to manage externals in a more homogeneous way in the future, and Pere liked the idea of using subtree for externals that we have copies in the repository. Let's add this topic to the group meeting today.

**Comment by Axel-Naumann** [comment]
I do not see a reason why this PR should probably be be blocked by your work on subtrees. Can't you deal with the zlib subtree after this has been merged, also given that the subtree idea is in the discussion stage? (I believe that's what Philippe meant with "tying this PR to that").

So - who's going to merge this? :-) @peremato ?

**Comment by amadio** [comment]
I wanted to have merged it as is before, but was asked not to. I am not blocking this either, as my first comment is from two weeks ago, while the PR is from June 19. Feel free to merge, we can deal with using git subtree later.

**Comment by pcanal** [comment]
@phsft-bot build please
